packages/dev/core/src/scene.ts

```diff
@@ -2944,22 +2937,17 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
     }
 
     /**
-     * Gets a the last added material using a given id
-     * @param id defines the material's Id
+     * Gets a last added material using a given id
+     * @param id defines the material's id
      * @param allowMultiMaterials determines whether multimaterials should be considered
      * @returns the last material with the given id or null if none found.
      */
     public getLastMaterialById(id: string, allowMultiMaterials: boolean = false): Nullable<Material> {
-        for (let index = this.materials.length - 1; index >= 0; index--) {
-            if (this.materials[index].id === id) {
-                return this.materials[index];
-            }
-        }
-        if (allowMultiMaterials) {
-            for (let index = this.multiMaterials.length - 1; index >= 0; index--) {
-                if (this.multiMaterials[index].id === id) {
-                    return this.multiMaterials[index];
-                }
+        const targetMaterials = allowMultiMaterials ? this.multiMaterials : this.materials;

```

Comment:
Same comment here: this won t work as it is not a OR situation. It is first looking in one, than the other one.

We can not concat here to prevent extra GC and generatops would require more tslib support unfortunately :-(